templates/default/mysql-job.sh.erb

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+# rdiff-backup MySQL job script for $JOB managed by Chef. Changes will be overwritten.
+
+# This script is normally run by cron, but can be run manually to start the job if it is not already running.
+
+FILE=`basename $0`
+DIR=`dirname $0`
+PIDFILE="$DIR/.$FILE.lock"
+JOB="${FILE%.sh}"
+LOGPATH='/var/log/rdiff-backup'
+INSTANCES=`pgrep -fu <%= @suser %> $FILE | wc -l`
+
+# Prepends date to every line of stdin, then writes to both stdout and the log file.
+function log() {
+  sed "s/^/[`date +'%Y-%m-%d %H:%M:%S'`] /" | tee >(/usr/sbin/cronolog "${LOGPATH}/${JOB}/%Y-%m.log")
+}
+
+# Prepends date to every line of stdin, then writes to stdout and both the individual and general log files.
+function logboth() {
+  sed "s/^/[`date +'%Y-%m-%d %H:%M:%S'`] /" | tee >(/usr/sbin/cronolog "${LOGPATH}/${JOB}/%Y-%m.log") >(/usr/sbin/cronolog "${LOGPATH}/general/%Y-%m-%d.log" -S "${LOGPATH}/current.log")
+}
+
+# If we're root instead of the rdiff-backup server user, rerun the script as it.
+if [ `whoami` = 'root' ]; then
+  su <%= @suser %> -c $0
+else
+
+  # Only run if this script isn't already running
+  if [ $INSTANCES -lt 3 ]; then
+    echo $$ > $PIDFILE 2> >(log)
+
+    echo '----------------------------------------' | log
+
+    echo "Backup starting for '$JOB'..." | logboth
+    mkdir -p "<%= @tempdest %>" 2>&1 | log
+    mkdir -p "<%= @dest %>" 2>&1 | log
+    nice -n 10 ssh -p <%= @port %> <%= @cuser %>@<%= @fqdn %> \
+      "mysqldump <%= @db %> -f -u <%= @muser %> -p'<%= @mpass %>' \
+      <% if @singletransaction %>
+      --single-transaction \
+      <% end %>" \
+      > <%= @tempdest %>/<%= @db %>.sql.part 2> >(log)
+
+    if [ $? -ne 0 ]; then
+      if [ -d <%= @dest %>/rdiff-backup-data ]; then
+        echo "Backup failed for '$JOB'." | logboth
+      else
+        echo "First backup failed for '$JOB'." | logboth
```

Comment:
Yes; individual Nagios checks are not created until the backup has run at least once, and since the summary check doesn't need to alert if a job fails that has its own check (so as not to be redundant), there needs to be a way to alert only if a job fails that does not have a check yet (i.e. on the first run).